internal/app/singularity/cache_clean_linux.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package singularity
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/sylabs/singularity/internal/pkg/client/cache"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+func cleanLibraryCache() error {
+	sylog.Debugf("Removing: %v", cache.Library())
+
+	err := os.RemoveAll(cache.Library())
+	if err != nil {
+		return fmt.Errorf("unable to clean library cache: %v", err)
+	}
+
+	return nil
+}
+
+func cleanOciCache() error {
+	sylog.Debugf("Removing: %v", cache.OciTemp())
+
+	err := os.RemoveAll(cache.OciTemp())
+	if err != nil {
+		return fmt.Errorf("unable to clean oci-tmp cache: %v", err)
+	}
+
+	return nil
+}
+
+func cleanBlobCache() error {
+	sylog.Debugf("Removing: %v", cache.OciBlob())
+
+	err := os.RemoveAll(cache.OciBlob())
+	if err != nil {
+		return fmt.Errorf("unable to clean oci-blob cache: %v", err)
+	}
+
+	return nil
+
+}
+
+// CleanCache : clean a type of cache (cacheType string). will return a error if one occurs.
+func CleanCache(cacheType string) error {
+	switch cacheType {
+	case "library":
+		err := cleanLibraryCache()
+		return err
+	case "oci":
+		err := cleanOciCache()
+		return err
+	case "blob", "blobs":
+		err := cleanBlobCache()
+		return err
+	case "all":
+		err := cache.Clean()
+		return err
+	default:
+		sylog.Fatalf("Not a valid type: %v", cacheType)
+		os.Exit(2)
+	}
+	return nil
+}
+
+func cleanLibraryCacheName(cacheName string) (bool, error) {
+	foundMatch := false
+	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
+	if err != nil {
+		return false, fmt.Errorf("unable to opening library cache folder: %v", err)
+	}
+	for _, f := range libraryCacheFiles {
+		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
+		if err != nil {
+			return false, fmt.Errorf("unable to look in library cache folder: %v", err)
+		}
+		for _, c := range cont {
+			if c.Name() == cacheName {
+				sylog.Debugf("Removing: %v", filepath.Join(cache.Library(), f.Name(), c.Name()))
+				err = os.RemoveAll(filepath.Join(cache.Library(), f.Name(), c.Name()))
```

Comment:
@bauerm97, I added back `c.Name()` here, because if you run: `singularity cache clean -T=library -N=ubuntu_latest.sif`, it would remove all `ubuntu_*.sif`.

But now it works fine, this may not be perfect, but it should be good enough...